src/main/fc/fc_rc.c

```diff
@@ -298,24 +308,36 @@ FAST_CODE uint8_t processRcSmoothingFilter(void)
             if (rxIsReceivingSignal() && (targetPidLooptime > 0) && (millis() > RC_SMOOTHING_FILTER_TRAINING_DELAY_MS)) {
                 rxFrameTimeSum += currentRxRefreshRate;
                 rxFrameCount++;
+                maxRxFrameInterval = MAX(maxRxFrameInterval, currentRxRefreshRate);
+                minRxFrameInterval = MIN(minRxFrameInterval, currentRxRefreshRate);
+                DEBUG_SET(DEBUG_RC_SMOOTHING, 0, rxFrameCount);           // log the step count during training
+                DEBUG_SET(DEBUG_RC_SMOOTHING, 3, currentRxRefreshRate);   // log each frame interval during training
                 if (rxFrameCount >= RC_SMOOTHING_FILTER_TRAINING_SAMPLES) {
-                    const float avgRxFrameRate = rxFrameTimeSum / rxFrameCount / 1000;
-                    defaultInputCutoffFrequency = calcRcSmoothingCutoff(avgRxFrameRate, RC_SMOOTHING_FILTER_INPUT_AUTO_HZ);
-                    defaultDerivativeCutoffFrequency = calcRcSmoothingCutoff(avgRxFrameRate, RC_SMOOTHING_FILTER_DERIVATIVE_AUTO_HZ);
+                    rxFrameTimeSum = rxFrameTimeSum - minRxFrameInterval - maxRxFrameInterval; // Throw out high and low samples
+                    calculatedFrameTimeAverageUs = lrintf(rxFrameTimeSum / (rxFrameCount - 2));
+                    const float avgRxFrameTimeMs = (rxFrameTimeSum / (rxFrameCount - 2)) / 1000;
```

Comment:
Why divide by 1000, then multiply by 1000 inside calculate?

Reply:
Oh, I see, your actually dividing twice to get seconds from microseconds. Just do it in one place by multiplying `1e-6f`